mlflow/server/handlers.py

```diff
@@ -1726,9 +1730,10 @@ def _validate_source(source: str, run_id: str) -> None:
             store = _get_tracking_store()
             run = store.get_run(run_id)
             source = pathlib.Path(local_file_uri_to_path(source)).resolve()
-            run_artifact_dir = pathlib.Path(local_file_uri_to_path(run.info.artifact_uri)).resolve()
-            if run_artifact_dir in [source, *source.parents]:
-                return
+            if is_local_uri(run.info.artifact_uri):
```

Comment:
If `run.info.artifact_uri` is not a local file URL, the following code `pathlib.Path(local_file_uri_to_path(run.info.artifact_uri)).resolve()` can still work, specifically, if input is a Http URL with *fragment* , it generates result of local file path which causes a vulnerability.

```
url = 'http://host/0010e54c8d9741b2ab83d2157dab66c1/artifacts#/../../../../../../../../../../../../../../etc/'

pathlib.Path(local_file_uri_to_path(run.info.artifact_uri)).resolve()  # get PosixPath('/private/etc') in MacOS.
```

Reply:
Looks good.